flask/app.py

```diff
@@ -1078,6 +1078,21 @@ def decorator(f):
             return f
         return decorator
 
+    @staticmethod
+    def _get_exc_class_and_code(exc_class_or_code):
+        """Ensure that we register only exceptions as handler keys"""
+        if isinstance(exc_class_or_code, integer_types):
+            exc_class = default_exceptions[exc_class_or_code]
+        else:
+            exc_class = exc_class_or_code
+        
+        assert issubclass(exc_class, Exception)
+        
+        if issubclass(exc_class, HTTPException):
```

Comment:
I think we missed this discussion: https://github.com/mitsuhiko/flask/pull/1291#discussion_r22427026

>

Why special-case HTTP exceptions and have a separate dict for each status code?

Since self.error_handler_spec is already backwards-incompatible (as in, code that directly modifies it breaks), I think we can remove this distinction. I suppose that also eliminates the need for most of the code in _ensure_exc_class.